object_goslice.go

```diff
@@ -349,3 +349,19 @@ func (o *objectGoSlice) sortGet(i int) Value {
 func (o *objectGoSlice) swap(i int, j int) {
 	(*o.data)[i], (*o.data)[j] = (*o.data)[j], (*o.data)[i]
 }
+
+func (o *objectGoSlice) MemUsage(ctx *MemUsageContext) (uint64, uint64, error) {
+	mem, newMem, err := o.baseObject.MemUsage(ctx)
+	if err != nil {
+		return 0, 0, err
+	}
+	for _, datum := range *o.data {
+		memValue, newMemValue, err := o.val.runtime.ToValue(datum).MemUsage(ctx)
```

Comment:
[q] this could be an expensive operation for large arrays. Should we estimate the usage instead? Example [here](https://github.com/mongodb-forks/goja/blob/1e62d9207652037afc12befe67cdefe0c394b660/array.go#L565). Same question for the `objectGoMapSimple`.

Reply:
Nathan just filed https://jira.mongodb.org/browse/BAAS-24348 which I will pick up pretty much right away!